### ISSUE TITLE
Add option to include pre bundle deps in the `application` builder

### ISSUE THREAD

**Issue by Carlosamouco** [angular/angular-cli#27091]
### Command

build, serve

### Description

In https://github.com/angular/angular-cli/pull/27021 an option to exclude dependencies from being pre bundled was added.
If we are trying to exclude a dependency that has nested common js dependencies, those should be pre bundled as stated in https://vitejs.dev/config/dep-optimization-options.html#optimizedeps-exclude.

### Describe the solution you'd like

Expose an option for including dependencies to be pre bundled similar to exclude option.

**Comment by clydin** [comment]
That statement from the Vite documentation doesn't apply to the Angular CLI's build system. With Angular, anything excluded from the prebundling will be bundled into the application code itself.  The module format is automatically handled by the underlying bundler (esbuild in this case).

**Comment by Carlosamouco** [comment]
@clydin thanks for the explanation. I still have a doubt, how should I proceeded with the nested dependencies of a dependency? Should I also exclude them then?

**Comment by clydin** [comment]
Can you provide the content of those warnings?

**Comment by Carlosamouco** [comment]
When I added a package to the exclude list, it gave me a warming for the nested commonjs dependencies saying the name of the dependency and that it could not be resolved and I was getting a error at runtime saying that it was not able to load chunk "X". I ended up adding also those libs to the exclude list and the problems went away. I can't remember what else I did and I can't reproduce it now so I must have done something wrong. 

On a off topic, it would be cool to support globs in the exclude list.
